app/models/post.rb

```diff
@@ -1,2 +1,25 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  after_create :update_posts_counter_on_create
+  after_destroy :update_posts_counter_on_destroy
```

Comment:
- [ ] It would good to add `after_save` to call the `#update_posts_counter` method every time a new post has been saved to database,

```
`after_save :update_posts_counter`
```